app/src/main/java/com/codrut/newsreader/view/bindings/RecyclerViewBinding.java

```diff
@@ -0,0 +1,35 @@
+package com.codrut.newsreader.view.bindings;
+
+import android.view.View;
+
+import androidx.databinding.BindingAdapter;
+import androidx.databinding.ObservableBoolean;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.codrut.newsreader.feature.newslist.adapter.NewsAdapter;
+import com.codrut.newsreader.feature.newslist.listener.NewsHandler;
+import com.codrut.newsreader.feature.newslist.model.ArticleItemViewModel;
+
+import java.util.List;
+
+public class RecyclerViewBinding {
+
+    @BindingAdapter({"items", "articleHandler"})
+    public static void setItems(RecyclerView recyclerView, List<ArticleItemViewModel> articles, NewsHandler handler) {
+        NewsAdapter adapter = (NewsAdapter) recyclerView.getAdapter();
+
+        if (adapter == null) {
+            adapter = new NewsAdapter(handler);
+            recyclerView.setLayoutManager(new LinearLayoutManager(recyclerView.getContext()));
+            recyclerView.setAdapter(adapter);
+        }
+
+        adapter.setItems(articles);
+    }
+
+    @BindingAdapter("visible")
+    public static void setVisible(RecyclerView recyclerView, ObservableBoolean value) {
```

Comment:
this binding adapter would be better suited into a more general ViewBindings class, as setVisibility is part of the super class View. This way we can reuse it in a more general way. Let's move this binding adapter to a  ViewBindings class and apply bind it to View (not only for RecyclerView - will still work for Recycler because Recycler is a View)